VideoLocker/res/layout/row_discussion_thread.xml

```diff
@@ -30,28 +33,35 @@
             android:layout_width="wrap_content"
             android:layout_height="wrap_content"
             android:textColor="@color/edx_grayscale_neutral_x_dark"
-            android:textSize="@dimen/edx_small" />
+            android:textSize="@dimen/edx_small"
+            android:maxLines="2"
+            android:ellipsize="end"
+            tools:text="This is a post title. It is a very long title to demonstrate that post titles will be truncated when they are too long."/>
 
         <LinearLayout
             android:orientation="horizontal"
             android:layout_width="wrap_content"
             android:layout_height="wrap_content">
 
-            <org.edx.mobile.third_party.iconify.IconView
-                android:id="@+id/discussion_post_pin_or_following_icon"
-                style="@style/icon_view_standard_size_style"
-                android:layout_marginEnd="3dp"
-                android:layout_marginRight="3dp"
-                app:iconColor="@color/edx_grayscale_neutral_light"
-                android:visibility="gone" />
+            <TextView
+                android:id="@+id/discussion_post_pin_text_view"
+                android:layout_width="wrap_content"
+                android:layout_height="wrap_content"
+                android:drawablePadding="3dp"
```

Comment:
> Why not? If the icon and text are twice as big, they would appear too close together unless the padding scales as well.

Right, but all of these changes might not necessarily be in the same proportion. My point was that when changing the layout, all the dimensions would probably need to be revised anyway. However, you can use `sp` units if you feel strongly about this.

> BTW, the current version of Iconify allows you to embed icons in a Textview via XML.

`FontTextView` was available when the library was originally imported, but was not included for some reason. However, we did used to use the icons in `TextView`, initializing them with the `Iconify.addIcons()` utility method. The problem with using the icons as part of the text was that we can't control it's size and alignment separate to the text. Also, I [implemented](#271) RTL auto-mirroring support in `IconDrawable` (which is used in `IconView`), but it can't be implemented properly in `TextView`.

> For some reason we have copied the iconify source code into the project, and an outdated version at that.

> (Most likely it wasn't available as a gradle dependency when we first used it)

It was available in Maven Central when it was imported, but Hanning used to import only the required classes from third-party libraries to keep the code lean. The [pull request](#230) contains the following comment:

> NOTE - i did not import the whole lib of android-iconify as it also includes duplicated font source file.

The library has now been updated to use version 4.4.0 of Font Awesome, but they seem to have   [reverted](JoanZapata/android-iconify@a628da6cc1d515c831526122740e1b46d5a376a5) my patch to add support for RTL auto-mirroring. I guess it was too much work to go through the new icons and decide which don't support RTL mirroring :disappointed: